### PR TITLE
Add sub method

### PR DESCRIPTION
**Pull request by andrewosh**
Extracted a revised version of `sub` from the batch parents PR.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -523,7 +555,7 @@ class Batch {
     let node = root = await this.getRoot()
     if (!node) node = root = TreeNode.create(null)
 
-    const seq = this.tree.feed.length + this.length
+    const seq = (this.tree._checkout ? this.tree._checkout : this.tree.feed.length) + this.length
```
Comment (mafintosh) [inline index.js:558]: this an unrelated bug fix?

Comment (andrewosh) [reply index.js:558]: ah, yes

**Comment by mafintosh** [comment]
1.0.0!
